core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigGenerationBuildStep.java

```diff
@@ -391,6 +392,17 @@ public void setupConfigOverride(
 
     @BuildStep
     public void watchConfigFiles(BuildProducer<HotDeploymentWatchedFileBuildItem> watchedFiles) {
```

Comment:
The watched files include external files like `.end` and `{user.dir}/config/application{-profile}.properties`. Should these be registered as a resource with the native image?

Reply:
Anything accessed through `java.lang.ClassLoader.getResources` (invoked by https://github.com/smallrye/smallrye-common/blob/c29bb1c00f484f5fa86f183e75c37af462ac6546/classloader/src/main/java/io/smallrye/common/classloader/ClassPathUtils.java#L84) needs to be registered (at least in theory).

Reply:
External files should be calling `processAsPath` https://github.com/smallrye/smallrye-common/blob/c29bb1c00f484f5fa86f183e75c37af462ac6546/classloader/src/main/java/io/smallrye/common/classloader/ClassPathUtils.java#L132-L160 directly, without using the classloader:

For `.env`:
https://github.com/smallrye/smallrye-config/blob/64295f6d4350d3b0922c69c093f1f106ba035b24/implementation/src/main/java/io/smallrye/config/DotEnvConfigSourceProvider.java#L54-L57

For `{user.dir}/config/application.properties`:
https://github.com/smallrye/smallrye-config/blob/e8d87ce1cfbf1e6f259a798c85498267dd12b32b/implementation/src/main/java/io/smallrye/config/AbstractLocationConfigSourceLoader.java#L118-L135

At least until now, we have never had issues loading these files and the native executable.

Reply:
Yes, these "issues" are popping up when using `-H:+ThrowMissingRegistrationErrors` (see https://github.com/quarkusio/quarkus/issues/41995) which aims to help developers detect unregistered accesses to elements. What I suspect is that there are many places where Quarkus tries to access unregistered elements and just fails without a warning, but since there are other places or ways to get the same data back it works as expected.

So in many of these cases it might be OK to not register the elements, but at the same time the best practice would be to ensure hat Quarkus also doesn't try to access them at run-time (e.g. through substitutions or changing configuration).

Reply:
@radcortez note that when the uri scheme is not defined smallrye tries both `tryFileSystem` and `tryClassPath` in https://github.com/smallrye/smallrye-config/blob/e8d87ce1cfbf1e6f259a798c85498267dd12b32b/implementation/src/main/java/io/smallrye/config/AbstractLocationConfigSourceLoader.java#L103-L104 with the latter accessing (or trying to) the config files through `java.lang.ClassLoader.getResources`.

If we don't want the external files to even be tried I think we should alter the value of `smallrye.config.locations`. WDYT?